### PR TITLE
ARROW-13624: [R] readr short type mapping has T and t backwards

### PR DESCRIPTION
**Pull request by dragosmg**

### CODE REVIEW

File: r/tests/testthat/test-csv.R [hunk 1]
```diff
@@ -329,3 +330,32 @@ test_that("Write a CSV file with invalid batch size", {
...
+    read_csv_arrow(tf,
+      col_names = c("dt", "time"),
+      col_types = "tT", skip = 1
+    )
+  )
+
+  expect_equal(
+    df,
+    tbl,
+    ignore_attr = TRUE
```
Comment (nealrichardson) [inline r/tests/testthat/test-csv.R:359]: Do we need to ignore all attributes?

```suggestion
    ignore_attr ="tzone"
```

File: r/tests/testthat/test-csv.R [hunk 2]
```diff
@@ -329,3 +330,32 @@ test_that("Write a CSV file with invalid batch size", {
...
+    col_types = "Tt",
+    skip = 1
+  )
+
+  expect_error(
+    read_csv_arrow(tf,
+      col_names = c("dt", "time"),
+      col_types = "tT", skip = 1
+    )
+  )
```
Comment (nealrichardson) [inline r/tests/testthat/test-csv.R:354]: Do we care to assert the error message here, or is it something from the C++ library that we don't want to write a test for?

Comment (dragosmg) [reply r/tests/testthat/test-csv.R:354]: I think the error message comes from the C++ library. The error message also references the CSV converter and parser using paths on my machine, so probably not useful to include as we would have to skip the test on CI.

Comment (thisisnic) [reply r/tests/testthat/test-csv.R:354]: > I think the error message comes from the C++ library

Yep, that's right.  I've run locally and pasted below.  One of the ways you can tell it's from the C++ library is that it starts `Error: Invalid:`

``` Error: Invalid: In CSV column #1: CSV conversion error to timestamp[s]: invalid value '16:20:00'
/home/nic2/arrow/cpp/src/arrow/csv/converter.cc:492  decoder_.Decode(data, size, quoted, &value)
/home/nic2/arrow/cpp/src/arrow/csv/parser.h:123  status
/home/nic2/arrow/cpp/src/arrow/csv/converter.cc:496  parser.VisitColumn(col_index, visit) 
```
> The error message also references the CSV converter and parser using paths on my machine, so probably not useful to include as we would have to skip the test on CI.

Technically, this wouldn't be a problem as we could just supply something like `regexp = "CSV conversion error to timestamp"` and not test for the full error.  However, as Neal says, it's a C++ error, so we probably don't want to test for it as the point here is testing the R package.

**Review by nealrichardson — APPROVED** [review]

**Pull request closed without merging**